routes/DeveloperApplication.test.ts

```diff
@@ -440,6 +440,86 @@ describe('validations', () => {
     });
   });
 
+  describe('internalApiInfo', () => {
+    const defaultInternalApiInfo = {
+      programName: 'Battle of the Hornburg',
+      sponsorEmail: 'aragorn@gondor.horse',
+      vaEmail: 'eowyn@va.gov',
+    };
+    describe('programName', () => {
+      it('is required', () => {
+        const internalApiInfo = { ...defaultInternalApiInfo, programName: undefined };
+        const payload = { ...defaultPayload, internalApiInfo };
+
+        const result = applySchema.validate(payload);
+
+        expect(result.error?.message).toEqual('"internalApiInfo.programName" is required');
+      });
+
+      it('is a string', () => {
+        const internalApiInfo = { ...defaultInternalApiInfo, programName: 12345 };
+        const payload = { ...defaultPayload, internalApiInfo };
+
+        const result = applySchema.validate(payload);
+
+        expect(result.error?.message).toEqual('"internalApiInfo.programName" must be a string');
+      });
+    });
+
+    describe('sponsorEmail', () => {
+      it('is required', () => {
+        const internalApiInfo = { ...defaultInternalApiInfo, sponsorEmail: undefined };
+        const payload = { ...defaultPayload, internalApiInfo };
+
+        const result = applySchema.validate(payload);
+
+        expect(result.error?.message).toEqual('"internalApiInfo.sponsorEmail" is required');
+      });
+
+      it('is a string', () => {
+        const internalApiInfo = { ...defaultInternalApiInfo, sponsorEmail: 12345 };
+        const payload = { ...defaultPayload, internalApiInfo };
+
+        const result = applySchema.validate(payload);
+
+        expect(result.error?.message).toEqual('"internalApiInfo.sponsorEmail" must be a string');
+      });
+
+      it('is in a valid format', () => {
+        const internalApiInfo = { ...defaultInternalApiInfo, sponsorEmail: 'lolnotanemail.com' };
+        const payload = { ...defaultPayload, internalApiInfo };
+
+        const result = applySchema.validate(payload);
+
+        expect(result.error?.message).toEqual(
+          '"internalApiInfo.sponsorEmail" must be a valid email',
+        );
+      });
+    });
+
+    describe('vaEmail', () => {
+      it('is a string', () => {
+        const internalApiInfo = { ...defaultInternalApiInfo, vaEmail: 12345 };
+        const payload = { ...defaultPayload, internalApiInfo };
+
+        const result = applySchema.validate(payload);
+
+        expect(result.error?.message).toEqual('"internalApiInfo.vaEmail" must be a string');
+      });
+
+      it('is in a valid format', () => {
+        const internalApiInfo = { ...defaultInternalApiInfo, vaEmail: 'lolnotanemail.com' };
```

Comment:
would it also be worth testing a valid email that doesn't end with `va.gov`?

Reply:
Good call out